x-pack/platform/plugins/shared/maps/ui_tests/tests/full_screen_mode.spec.ts

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { expect, tags, test } from '@kbn/scout';
+import {
+  VISIBLE_CHROME,
+  HIDDEN_CHROME,
+  FULL_SCREEN_MODE,
+  EXIT_FULL_SCREEN,
+} from '../fixtures/constants';
+
+test.describe('Maps', { tag: tags.DEPLOYMENT_AGNOSTIC }, () => {
+  test.beforeEach(async ({ browserAuth, pageObjects }) => {
+    await browserAuth.loginAsViewer();
+    await pageObjects.maps.goto();
```

Comment:
The Maps application shows different views based on saved objects
* when users navigate to the application with no map saved objects - application shows new map
* when users navigate to application with map saved objects - application shows map listing page.

Original GisPage worked around this ambiguity with methods like `openNewMap` and `gotoMapListingPage`. I think this level of specificity is required for functional testing and `goto` should be replaced by `openNewMap`.

Reply:
Ah ok, so I appreciate that fact. 
But the test that is added in this pr, doesn't support any of that.  

So, how about we address this in the next pr that adds a test that needs this?
Wdyt?

Reply:
I just think a generic `goto` function will result in flaky tests, since the resolved URL changes based on state of the system. 

I am fine leaving as is as long as `goto` is marked as deprecated with a comment that `goto` is to be replaced with specific methods `openNewMap` and `gotoMapListingPage`.

Reply:
that works for me.  again, thanks for the feedback man!